tools/pylib/boutdata/data.py

```diff
@@ -247,6 +248,44 @@ def __init__(self, filename, name="root"):
 
                         section[line[:eqpos].strip()] = value
 
+    def write(self, filename=None, overwrite=False):
+        """ Write to BOUT++ options file
+
+        This method will throw an error rather than overwriting an existing
+        file unless the overwrite argument is set to true.
+        Note, no comments from the original input file are transferred to the
+        new one.
+
+        Parameters
+        ----------
+        filename : str
+            Path of the file to write
+            (defaults to path of the file that was read in)
+        overwrite : bool
+            If False then throw an exception if 'filename' already exists.
+            Otherwise, just overwrite without asking.
+            (default False)
+        """
+        if filename is None:
+            filename = self.filename
+
+        if not overwrite and os.path.exists(filename):
+            raise ValueError("Not overwriting existing file, cannot write output to "+filename)
+
+        def write_section(basename, opts, f):
+            if basename is not "":
```

Comment:
Could just be `if basename:`, unless I'm missing something? Similarly for the same conditional below.